modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -209,6 +265,23 @@ static GCompileArg exec(cv::gapi::s11n::IIStream& is, const std::string& tag) {
 }
 };
 
+template<typename T> struct deserialize_runarg;
+
+template<typename RMatAdapterType>
+struct deserialize_runarg {
+static GRunArg exec(cv::gapi::s11n::IIStream& is, uint32_t idx) {
+    if (idx == GRunArg::index_of<RMat>()) {
+        RMatAdapterType adapter = cv::gapi::s11n::detail::S11N<RMatAdapterType>::deserialize(is);
+        RMat rmat = make_rmat<RMatAdapterType>(adapter);
```

Comment:
I thought it should be something like

```c++
auto ptr = make_shared<T>();
ptr->deserialize(os);
return RMat(std::move(ptr);
```

Reply:
In that case we need a default constructor for adapter class. Is this ok? Switched to the proposed approach though

Reply:
Yes, this is ok.